### PR TITLE
Update docker.md

### PR DESCRIPTION
**Pull request by fsilvaortiz**
Git and node-gyp installation is added. Without these, errors occur when trying to create the image.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by awanlin** [comment]
Hi @fsilvaortiz, thanks for this contributions but the solution in this PR is probably the better way for us to handle this: https://github.com/backstage/backstage/pull/20695. What do you think?

**Comment by fsilvaortiz** [comment]
Hi @awanlin, that PR fixes the missing node-gyp package and it's a better solution. I added the git package only to address this warning:

 [build 8/9] RUN yarn --cwd packages/backend build                                                                                                                                                                                                                                                                                                      8.1s
 => => # app:  WARNING: Failed to read git commit, ExitCodeError: Command 'git rev-parse HEAD' exited with code 127
 => => # app:
 => => # app:  /bin/sh: 1: git: not found
 => => # app:
 => => # app:  WARNING: Failed to describe git version, ExitCodeError: Command 'git describe --always' exited with code 127

**Comment by awanlin** [comment]
As that output from git is just warnings and not actually errors that impact functionality I'm not sure it needs to be added. But if we do add this there are several other places that should be updated as well

@freben @vinzscam what do you think?

**Comment by freben** [comment]
Agreed, just merged #20695 now for the node-gyp change.

Regarding the git warning - am I right in understanding that you are performing the multi-stage build, so you do the build step inside a docker image? @awanlin is correct that in a sense it doesn't affect runtime much, except that it does make it possible to inspect the built commit and version in the interface later on. In the spirit of wanting to have the dockerized build have the same features as the host build, I don't mind much adding git there but only there.

So if you scale this PR down to only the git change, I think we're good 👍

**Comment by fsilvaortiz** [comment]
Hi! I undid the change of node-gyp installation and left only the git one.

File: docs/deployment/docker.md [hunk 1]
```diff
@@ -226,7 +226,7 @@ RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
     --mount=type=cache,target=/var/lib/apt,sharing=locked \
     apt-get update && \
     apt-get install -y --no-install-recommends libsqlite3-dev
-
+
```
Comment (awanlin) [inline docs/deployment/docker.md:229]: Can we remove this extra space, please?

**Review by awanlin — CHANGES_REQUESTED** [review]
Looks good just one small nit then I think we are good 👍

**Review by awanlin — APPROVED** [review]
Awesome! Thanks @fsilvaortiz 🚀

**Comment by awanlin** [comment]
Hi @freben, can you give this a quick look, please? It's a small change, looks ready to merge from my perspective 👍

**Comment by freben** [comment]
Build is still red though. Something is making prettier angry - the second line looks unexpected right?

**Comment by awanlin** [comment]
> Build is still red though. Something is making prettier angry - the second line looks unexpected right?

Shoot, missed that, thought it was just the reoccurring microsite issues :(

@fsilvaortiz can you take a look at this, please?

**Comment by freben** [comment]
Hey sorry, I see DCO is failing too - replacing this with #21348, thanks for raising this 🙏

**Pull request closed without merging**